src/libcrun/container.c

```diff
@@ -1187,6 +1187,13 @@ container_init_setup (void *args, pid_t own_pid, char *notify_socket, int sync_s
   if (clearenv ())
     return crun_make_error (err, errno, "clearenv");
 
+  // set primary process to 1 explicitly if nothing is configured and LISTEN_FD is not set
+  if (getenv ("LISTEN_PID") == NULL && getenv ("LISTEN_FDS") == NULL && entrypoint_args->context->preserve_fds > 0)
+    {
+      setenv ("LISTEN_PID", "1", 1);
```

Comment:
is `getenv ("LISTEN_FDS") == NULL` needed?

Reply:
@giuseppe no this is not needed anymore removed it in latest commit. But keeping `getenv("LISTEN_PID") == NULL` just to honor user's PID config.